CloudNotes/CloudNotes/Sources/ViewControllers/MemoViewController.swift

```diff
@@ -0,0 +1,202 @@
+        guard false == memo.isTitleEmpty else {
+            return textView.text = nil
+        }
+
+        textView.text = "\(memo.title)\n\(memo.body)"
+    }
+
+    private func resetScrollOffset() {
+        let topOffset = CGPoint(x: 0, y: -view.safeAreaInsets.top)
+        textView.setContentOffset(topOffset, animated: false)
+    }
+
+    // MARK: Keyboard observing
+
+    @objc private func textViewMoveUp(_ notification: NSNotification) {
+        guard let userInfo = notification.userInfo,
+              let keyboardFrame = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect else { return }
+
+        let contentInset = UIEdgeInsets(top: 0, left: 0, bottom: keyboardFrame.size.height, right: 0)
+        textView.contentInset = contentInset
+        textView.scrollIndicatorInsets = contentInset
+    }
+
+    @objc private func textViewMoveDown() {
+        let contentInset = UIEdgeInsets.zero
+        textView.contentInset = contentInset
+        textView.scrollIndicatorInsets = contentInset
+    }
+
+    // MARK: Method
+
+    func textViewResignFirstResponder() {
+        textView.resignFirstResponder()
+    }
+
+    func setTextViewHidden(is value: Bool) {
+        textView.isHidden = value
+    }
+
+}
+
+// MARK: - UISplitViewControllerDelegate
+
+extension MemoViewController: UISplitViewControllerDelegate {
+
+    func splitViewController(_ svc: UISplitViewController,
+                             topColumnForCollapsingToProposedTopColumn proposedTopColumn: UISplitViewController.Column) -> UISplitViewController.Column {
+        return textView.isHidden ? .primary : .secondary
+    }
+
+}
+
+// MARK: - UITextViewDelegate
+
+extension MemoViewController: UITextViewDelegate {
+
+    func textViewDidChange(_ textView: UITextView) {
+        guard let textViewText = textView.text,
+              let seperatedMemo = separatedMemo(from: textViewText) else { return }
+
+        self.memo = Memo(title: seperatedMemo.newTitle, body: seperatedMemo.newBody)
+
+        let primaryViewController = splitViewController?.viewController(for: .primary) as? MemoListViewController
+
+        if let row = row,
+           let memo = memo {
+            primaryViewController?.updateMemo(at: row, to: memo)
+            self.row = Style.updatedMemoRow
+        }
+    }
+
+    private func separatedMemo(from text: String) -> (newTitle: String, newBody: String)? {
+        let separatedText: [Substring] = text.split(separator: Style.memoSeparator, maxSplits: 1, omittingEmptySubsequences: true)
+
+        var newTitle = String()
+        var newBody = String()
+
+        switch separatedText.count {
+        case 0:
+            break
+        case 1:
+            newTitle = String(separatedText[0])
+        case 2:
+            newTitle = String(separatedText[0])
+            newBody = String(separatedText[1])
+        default:
+            return nil
+        }
+
+        return (newTitle, newBody)
+    }
```

Comment:
좀 더 간단히 쓸 수도 있을 것 같아요 ㅎㅎ
```suggestion
    private func separatedMemo(from text: String) -> (newTitle: String, newBody: String)? {
        let separatedText: [Substring] = text.split(separator: Style.memoSeparator, maxSplits: 1, omittingEmptySubsequences: true)
    return (String(separatedText.first ?? ""), String(separatedText.last ?? ""))
    }
```

Reply:
오.. 제가 너무 돌아서 가고 있었군요! 좋은 코드 제시 감사합니다 :)